omod/src/main/java/org/openmrs/module/reportingrest/web/resource/ReportDesignResource.java

```diff
@@ -0,0 +1,125 @@
+ */
+@Resource(name = RestConstants.VERSION_1 + ReportingRestController.REPORTING_REST_NAMESPACE + "/reportDesign",
+    supportedClass = ReportDesign.class, supportedOpenmrsVersions = {"1.8.* - 9.9.*"})
+public class ReportDesignResource extends DelegatingCrudResource<ReportDesign> {
+
+  @Override
+  public ReportDesign newDelegate() {
+    return new ReportDesign();
+  }
+
+  @Override
+  public ReportDesign getByUniqueId(String uuid) {
+    return getReportService().getReportDesignByUuid(uuid);
+  }
+
+  @Override
+  protected PageableResult doSearch(RequestContext context) {
+    String reportDefinitionUuid = context.getParameter("reportDefinitionUuid");
+    if (StringUtils.isBlank(reportDefinitionUuid)) {
+      throw new IllegalArgumentException("reportDefinitionUuid parameter is required");
+    }
+
+    ReportDefinition reportDefinition = getReportDefinitionService().getDefinitionByUuid(reportDefinitionUuid);
+    if (reportDefinition == null) {
+      return new EmptySearchResult();
+    }
+
+    List<ReportDesign> reportDesigns = getReportService().getReportDesigns(reportDefinition, null, false);
+    return new NeedsPaging<ReportDesign>(reportDesigns, context);
+  }
+
+  @Override
+  protected void delete(ReportDesign reportDesign, String reason, RequestContext requestContext) throws ResponseException {
+    purge(reportDesign, requestContext);
+  }
+
+  @Override
+  public ReportDesign save(ReportDesign reportDesign) {
+    return getReportService().saveReportDesign(reportDesign);
+  }
+
+  @Override
+  public void purge(ReportDesign reportDesign, RequestContext requestContext) throws ResponseException {
+    if (reportDesign != null) {
+      getReportService().purgeReportDesign(reportDesign);
+    }
+  }
+
+  @Override
+  public DelegatingResourceDescription getCreatableProperties() {
+    return new DelegatingResourceDescription();
+  }
+
+  @Override
+  public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+    DelegatingResourceDescription description = null;
+
+    if (representation instanceof RefRepresentation) {
+      description = new DelegatingResourceDescription();
+      description.addProperty("uuid");
+      description.addProperty("name");
+      description.addProperty("rendererType");
+      description.addSelfLink();
+    } else if (representation instanceof FullRepresentation) {
+      description = new DelegatingResourceDescription();
+      description.addProperty("uuid");
+      description.addProperty("name");
+      description.addProperty("rendererType");
+      description.addSelfLink();
+      description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+    } else if (representation instanceof DefaultRepresentation) {
+      description = new DelegatingResourceDescription();
+      description.addProperty("uuid");
+      description.addProperty("name");
+      description.addProperty("rendererType");
+      description.addSelfLink();
+    }
```

Comment:
If these aren't needed to be different, I would just do something like:

```
if (rep instanceof CustomRepresentation) {
  return null;
}
DelegatingResourceDescription description = new DelegatingResourceDescription();
description.addProperty("uuid");
description.addProperty("name");
description.addProperty("rendererType");
description.addSelfLink();
return description;
```

Reply:
corrected